src/lib/Types/Core.hs

```diff
@@ -153,7 +153,11 @@ data Block n where
   Block :: BlockAnn n l -> Nest Decl n l -> Atom l -> Block n
 
 data BlockAnn n l where
-  BlockAnn :: Type n -> BlockAnn n l
+  -- Make the EffectRow lazy because it requires a traversal to
+  -- compute, and there are many places in the code that ignore it.
+  -- (Also, it's a S.Set, so isn't obviously much smaller than the
+  -- thunk.)
+  BlockAnn :: Type n -> ~(EffectRow n) -> BlockAnn n l
```

Comment:
Since there doesn't seem to be any significant difference in your benchmarks, we could just make it eager for simplicity.

Reply:
Done.